### PR TITLE
add run_ssh_command

### PR DESCRIPTION
**Pull request by dbasunag**

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: pyhelper_utils/shell.py
Comment (myakove) [inline pyhelper_utils/shell.py:76]: missing typing

**Review by myakove — CHANGES_REQUESTED** [review]

File: pyhelper_utils/shell.py
Comment (dbasunag) [reply pyhelper_utils/shell.py:76]: I will add that.

File: pyhelper_utils/shell.py
Comment (dbasunag) [reply pyhelper_utils/shell.py:88]: Ok. Will do.

File: tests/test_shell.py
Comment (dbasunag) [reply tests/test_shell.py:53]: This is the simplest way to mock a function return within a context, without this it is hard to read and understand the mocking part.

File: pyhelper_utils/shell.py
Comment (dbasunag) [reply pyhelper_utils/shell.py:89]: I have added a section with example.

File: README.md
Comment (myakove) [inline README.md:38]: Please remove this `run_ssh_command():` this is just how to call function in python

File: pyhelper_utils/shell.py
Comment (myakove) [inline pyhelper_utils/shell.py:93]: can you use `shlex` in the example

**Review by myakove — CHANGES_REQUESTED** [review]

File: README.md
Comment (dbasunag) [reply README.md:38]: Ok. Do you want me to remove the line `Enables running a command against a remote server over SSH. It expects a host object, created using [python-rrmngmnt](https://github.com/rhevm-qe-automation/python-rrmngmnt).
` as well? and only keep the sample code?

File: pyproject.toml
Comment (rnetser) [inline pyproject.toml:58]: should not be under `test` dependency

**Review by rnetser — CHANGES_REQUESTED** [review]

File: tests/test_shell.py
Comment (rnetser) [inline tests/test_shell.py:43]: please place fixtures before the tests in the file

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by myakove — CHANGES_REQUESTED** [review]

File: pyhelper_utils/shell.py
Comment (dbasunag) [reply pyhelper_utils/shell.py:107]: I don't understand this suggestion. Can you please explain why this is needed?

File: pyhelper_utils/shell.py
Comment (dbasunag) [reply pyhelper_utils/shell.py:108]: Same here, I don't understand why this is needed.

File: pyhelper_utils/shell.py
Comment (myakove) [reply pyhelper_utils/shell.py:107]: the suggestion is to add type to your list, what is not understood?

Comment (dbasunag) [reply pyhelper_utils/shell.py:107]: Typing should be on docs, no? Do we now need to use it for any statements?

Comment (dbasunag) [reply pyhelper_utils/shell.py:107]: @myakove I found https://realpython.com/python-type-checking/#variable-annotations, I still don't understand how this is more beneficial. But let me commit the suggestions and test it out.

File: pyhelper_utils/shell.py
Comment (dbasunag) [reply pyhelper_utils/shell.py]: This had to be undone, because of mypy failing ```pyhelper_utils/shell.py:108: error: Name "commands" already defined on line 80  [no-redef]
Found 1 error in 1 file (checked 6 source files)
```

File: pyhelper_utils/exceptions.py
Comment (myakove) [inline pyhelper_utils/exceptions.py:5]: please use 

`from __future__ import annotations`  >> must be import at first line.
`str | None = None`

**Review by myakove — CHANGES_REQUESTED** [review]

File: pyhelper_utils/exceptions.py
Comment (dbasunag) [reply pyhelper_utils/exceptions.py:5]: Ok, I see what you have done in a previous PR. Will do

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Comment by dbasunag** [comment]
rebased after approval. Merging.

**Pull request merged**
